pymeasure/instruments/keysight/keysightN7776C.py

```diff
@@ -0,0 +1,256 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+import numpy as np
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+WL_RANGE = [1480,1620]
+LOCK_PW = 1234
+
+class KeysightN7776C(Instrument):
+    """
+    This represents the Keysight N7776C Tunable Laser Source interface.
+    .. code-block:: python
+        laser = N7776C( address )
+        laser.sweep_wl_start = 1550
+        laser.sweep_wl_stop = 1560
+        laser.sweep_speed = 1
+        laser.sweep_mode = 'CONT'
+        laser.output = 1
+        while laser.sweep == 1:
+            continue
+        laser.output = 0
+    
+    """
+    def __init__(self, address, **kwargs):
+        super(KeysightN7776C, self).__init__(
+            address, "N7776C Tunable Laser Source",**kwargs)
+
+    def reset(self):
+        """
+        Reset the laser controller to its default state.
+        """
+        self.write('*RST')
+
+    def set_power(self,value,unit='mW'):
+        """
+        Function wrapper to set the power in an arbitrary unit
+        """
+        if not unit in ['dBm','mW']:
+            raise ValueError('Unknown Power Unit.')
+
+        self._output_power_unit = unit
+        if unit == 'mW':
+            self._output_power = value*1e-3 #Conversion to mW
```

Comment:
so the device has a power setting "mW", but then wants the value to be sent in "W"? Alternatively, for a conversion _to mW_ (as your comment indicates) with *1e-3, `value` has to be in µW - I have to admit I'm confused by this.

I would expect with such a signature, that an `instr.set_power(35, 'mW')` sets a power of 0.035 W (i.e. 35 mW) - in that context I don't know why you divide by 1000 again, or what unit the device expects the power in (if not "mW" as its `_output_power_unit` is being set).